File: src/test/java/io/tripled/social/client/presentation/controller/WallControllerTest.java
package io.tripled.social.client.presentation.controller;

import io.tripled.social.client.application.SpiedWallMessageApplicationService;
import io.tripled.social.client.presentation.TestOutput;
import io.tripled.social.client.presentation.request.WallCliRequest;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.equalTo;

class WallControllerTest {

  private WallController wallController;
  private SpiedWallMessageApplicationService service;

  @BeforeEach
  void setUp() {
    service = new SpiedWallMessageApplicationService();
    wallController = new WallController(service);
  }

  @Test
  void executesCommand() {
    WallCliRequest request = new WallCliRequest("Alice");

    TestOutput output = new TestOutput();
    wallController.execute(request, output);

    assertThat(service.userName, equalTo("Alice"));
    output.assertContains(equalTo("Alice"));
  }

}

File: src/test/kotlin/io/tripled/social/client/presentation/controller/WallControllerTest.kt
package io.tripled.social.client.presentation.controller

import io.tripled.social.client.application.SpiedReadWallUseCase
import io.tripled.social.client.presentation.TestOutput
import io.tripled.social.client.presentation.request.WallCliRequest
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class WallControllerTest {
    private var wallController: WallController? = null
    private var service: SpiedReadWallUseCase? = null

    @BeforeEach
    fun setUp() {
        service = SpiedReadWallUseCase()
        wallController = WallController(service!!)
    }

    @Test
    fun executesCommand() {
        val request = WallCliRequest("Alice")

        val output = TestOutput()
        wallController!!.execute(request, output)

        MatcherAssert.assertThat<String?>(service!!.userName, Matchers.equalTo<String?>("Alice"))
        output.assertContains(Matchers.equalTo<String?>("Alice"))
    }
}
